fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -303,6 +309,22 @@ ACTOR Future<Void> trackShardMetrics(DataDistributionTracker::SafeAccessor self,
 					if (shardMetrics->get().present()) {
 						self()->dbSizeEstimate->set(self()->dbSizeEstimate->get() + metrics.first.get().bytes -
 						                            shardMetrics->get().get().metrics.bytes);
+						if (SERVER_KNOBS->SHARD_ENCODE_LOCATION_METADATA && SERVER_KNOBS->ENABLE_DD_PHYSICAL_SHARD) {
+							// update physicalShard metrics and return whether the keys needs to move out of
+							// physicalShard
+							bool needToMove = self()->physicalShardCollection->trackPhysicalShard(
+							    keys, metrics.first.get(), shardMetrics->get().get().metrics, initWithNewMetrics);
+							if (needToMove) {
+								// Do we need to update shardsAffectedByTeamFailure here?
+								self()->output.send(
+								    RelocateShard(keys,
```

Comment:
This assumes there are multiple keyranges in a physical shard? Do we split a physical shard when there is only one keyrange in the shard?

Reply:
No, we won't split. The current DD will split it.

Reply:
that's why when we decide whether to move out a shard due to the too large size of physicalShard, we check whether this shard is the only keyRange in the physicalShard.